frontend-react/src/utils/RSError.ts

```diff
@@ -0,0 +1,55 @@
+export type ErrorUI = "message" | "page";
+export const ERROR_UI_DEFAULT: ErrorUI = "message";
+/** For consistency, when passing the code prop, please use these values
+ * e.g. <ErrorComponent code={RSError.NOT_FOUND} /> */
+export enum ErrorName {
+    // TODO: Update App.tsx to throw on bad browser, wrap with boundary in index.ts?
+    UNSUPPORTED_BROWSER = "unsupported-browser",
+    UNAUTHORIZED = "unauthorized",
+    NOT_FOUND = "not-found",
+    // Any error thrown that cannot be parsed by RSError.parseStatus()
+    UNKNOWN = "unknown-error",
+    // Any error that does not extend the RSError class
+    NON_RS_ERROR = "non-rs-error",
+}
+
+/** All custom errors should extend this class */
+export abstract class RSError extends Error {
+    /* Used for identifying unique content to display for error */
+    code: ErrorName;
+    /* Used to determine if this error should render as a message or full page */
+    displayAs: ErrorUI = ERROR_UI_DEFAULT;
+
+    /** @param message {string} Passed to `Error.constructor()`
+     * @param status {number?} Used to parse `RSError.code` value
+     * @param display {ErrorUI} Used to display as either a message or page */
+    protected constructor(message: string, status?: number, display?: ErrorUI) {
+        super(message); // Sets message
+        this.code = this.parseStatus(status); // Sets code using child's parseStatus
```

Comment:
The thought originally was that there's be other blanket types, say an `RSFormFieldError`, that takes in some kind of indicator of _what kind of_ field on the form has an error -- or maybe even at a lower level it's a character limit error or an invalid email error. 

The alternative (which I _do not_ like) for single-type errors would be just returning the right ErrorName from that method. I put that in the docs, but I'm open to ideas on both if we need, and how we can cleaner implement that detailed approach to erroring. The immediate use that came to mind is table filters and the admin settings forms since there's so many unique rules built in to how those are supposed to be shaped, and I don't _think_ we have a system validating those other than the back-end saying "hey this was invalid".

Reply:
I think I was working ahead a bit to help generalize the ability to parse different types of errors within a class. I'm gonna take a look at this, because I'm also having some second thoughts on other parts of the PR. I might refine the approach a bit more and clean this up in the process